### DISCUSSION TITLE
"Unselect All" button for Data Grid columns selector

### DISCUSSION THREAD

**Discussion by ilyabakharev** [vmware-clarity/ng-clarity#464] [Ideas]
Add the "Unselect All" button that will remove all the columns from the Data Grid

![image](https://user-images.githubusercontent.com/1030919/211919166-3c406721-40f4-4848-8c97-60fba371eaa9.png)

**Comment by kevinbuhmann** [comment]
What is the use-case here? All columns cannot be deselected. There must be at least one selected column.

**Reply by ilyabakharev** [reply]
When you have some required columns and some optional.  So, columns A, B, and C are always shown (no clrDgHideableColumn attribute) and columns D, E, and F are optional (hidden: false).  User clicks select all and later decides to unselect all.

**Reply by kevinbuhmann** [reply]
Okay, I see. So would it make sense to only show "unselect all" if there are columns that cannot be hidden? I'm asking because all columns can be hidable, but the datagrid won't let you deselect all columns in that case.

**Reply by kevinbuhmann** [reply]
Actually, if all columns are marked hidable, we could either 1) not show the "unselect all" option or 2) deselect all except for the first column when "unselect all" is activated. Maybe some other option?

**Reply by ilyabakharev** [reply]
I think that's a fair compromise.  If all columns can hidden, the unselect all would be invalid (or would unselect all except the first column).

**Reply by kevinbuhmann** [reply]
cc: @pdaigle
